core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -736,30 +736,31 @@ private[spark] class Master(
     val appName = app.desc.name
     val notFoundBasePath = HistoryServer.UI_PATH_PREFIX + "/not-found"
     try {
-      val eventLogFile = app.desc.eventLogDir
-        .map { dir => EventLoggingListener.getLogPath(dir, app.id) }
+      val eventLogDir = app.desc.eventLogDir
         .getOrElse {
           // Event logging is not enabled for this application
           app.desc.appUiUrl = notFoundBasePath
           return false
         }
-        
-      val fs = Utils.getHadoopFileSystem(eventLogFile, hadoopConf)
-
-      if (fs.exists(new Path(eventLogFile + EventLoggingListener.IN_PROGRESS))) {
+      
+      val eventLogFilePrefix = EventLoggingListener.getLogPath(eventLogDir, app.id)        
+      val fs = Utils.getHadoopFileSystem(eventLogDir, hadoopConf)
+      val eventLogFileSuffix = if (fs.exists(new Path(eventLogFilePrefix + 
+          EventLoggingListener.IN_PROGRESS))) {
         // Event logging is enabled for this application, but the application is still in progress
-        val title = s"Application history not found (${app.id})"
-        var msg = s"Application $appName is still in progress."
+        var msg = s"Application $appName is still in progress, it may be terminated accidently."
         logWarning(msg)
-        msg = URLEncoder.encode(msg, "UTF-8")
-        app.desc.appUiUrl = notFoundBasePath + s"?msg=$msg&title=$title"
-        return false
-      }
+        EventLoggingListener.IN_PROGRESS
+      } else ""
```

Comment:
Like:

`val inProgressExists = fs.exists(new Path(eventLogFilePrefix + EventLoggingListener.IN_PROGRESS))`

Then use that in the `if` statement, which need not also evaluate to a value.
Then use the same flag to construct the file name.
IMHO this is more straightforward than piggy-backing a sometimes-blank return value from the conditional.